framework/packages/abstract-ica/src/action.rs

```diff
@@ -0,0 +1,81 @@
+use cosmwasm_std::{Binary, Coin, CosmosMsg};
+
+/// Interchain Account Action
+#[cosmwasm_schema::cw_serde]
+#[non_exhaustive]
+pub enum IcaAction {
+    // Execute on the ICA
+    Execute(IcaExecute),
+    // Query on the ICA
+    // Query(IcaQuery),
+    // Send funds to the ICA
+    Fund {
+        funds: Vec<Coin>,
+        // Optional receiver address
+        // Should be formatted in expected formatting
+        // EVM: HexBinary
+        // Cosmos: Addr
+        receiver: Option<Binary>,
+        memo: Option<String>,
+    },
+    // ... other actions?
+}
+
+impl IcaAction {
+    // Used to set ordering
+    pub fn discriminant(&self) -> u8 {
+        match self {
+            IcaAction::Execute(_) => 0,
+            IcaAction::Fund { .. } => 1,
+            // IcaAction::Query(_) => 2,
+        }
+    }
+}
+
+/// Queries first
+/// Execute second
+/// Funds transfers last
+impl PartialOrd for IcaAction {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
```

Comment:
I see 3 definitions for order here:
1) Execute first, Fund transfers second, Query last
2) Queries first, Execute second, Funds transfers last
(third will be in a next comment)

Reply:
Removed sorting